### PR TITLE
Update codebase to comply with up-spec 1.5.8 L1

### PR DESCRIPTION
**Pull request by neelam-kushwah**
In version 1.5.8, we have implemented the following enhancements:

- **URI**: Unified definitions into a single format, eliminating the long and micro forms. Predicates have been simplified for improved usability.
- **uTransport API**: Enhanced to support pattern matching for all message types.
- **UAttributes**: Updated the data model and removed UPayload, with the format now in the header and payload data directly in UMessage.
- **uProtocol Custom Options**: Moved to uoptions.proto to include SOME/IP serialization support and explicit topic declaration.
-  **Remove Cloudevent code**: Not needed in up-python
- #40

### CODE REVIEW

**Comment by neelam-kushwah** [comment]
@princemac, Would you mind taking a look at it? It has major changes, so I would really appreciate your feedback.

File: scripts/pull_and_compile_protos.py [hunk 1]
```diff
@@ -1,22 +1,14 @@
 """
-SPDX-FileCopyrightText: Copyright (c) 2023 Contributors to the
-Eclipse Foundation
+SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
```
Comment (matthewd0123) [inline scripts/pull_and_compile_protos.py:2]: Not super important, but according to @stevenhartley and eclipse, the copyright date should be when the file was created, not edited. So, if this file was originally merged in 2023, copyright date should stay 2023.

File: uprotocol/transport/builder/umessagebuilder.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+            .with_reqid(request.id)
+        )
+
+    def __init__(self, source: UUri, id_val: UUID, type_val: UMessageType):
+        """Construct the UMessageBuilder with the configurations that are
+        required for every payload transport.
+
+        @param source   Source address of the message.
+        @param id       Unique identifier for the message.
+        @param type     Message type such as Publish a state change,
```
Comment (matthewd0123) [inline uprotocol/transport/builder/umessagebuilder.py:156]: type_val, not type. the variable used to be called "type", but that shadowed the built-in python "type", so it had to change

File: uprotocol/transport/builder/umessagebuilder.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+            .with_sink(request.source)
+            .with_reqid(request.id)
+        )
+
+    def __init__(self, source: UUri, id_val: UUID, type_val: UMessageType):
+        """Construct the UMessageBuilder with the configurations that are
+        required for every payload transport.
+
+        @param source   Source address of the message.
+        @param id       Unique identifier for the message.
```
Comment (matthewd0123) [inline uprotocol/transport/builder/umessagebuilder.py:155]: id_val, not id. the variable used to be called "id", but that shadowed the built-in python "id", so it had to change.

File: uprotocol/transport/validator/uattributesvalidator.py [hunk 4]
```diff
@@ -343,10 +361,24 @@ def validate_req_id(self, attributes_value: UAttributes) -> ValidationResult:
...
+            return ValidationResult.failure("Missing correlationId")
+        if not UUIDUtils.is_uuid(attributes_value.reqid):
+            return ValidationResult.failure("Invalid correlation UUID")
+        return ValidationResult.success()
+
+    def validate_priority(self, attributes_value: UAttributes) -> ValidationResult:
+        """
+        Validate the priority value to ensure it is one of the known CS values
+
+        @param attributes Attributes object containing the Priority to validate.
```
Comment (matthewd0123) [inline uprotocol/transport/validator/uattributesvalidator.py:374]: either this should be attributes_value or arg should be attributes

File: uprotocol/uri/factory/uri_factory.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    URI Factory that builds URIs from protos
+    """
+
+    @staticmethod
+    def from_proto(
+        service_descriptor: Optional[ServiceDescriptor], resource_id: int, authority_name: Optional[str]
+    ) -> UUri:
+        """
+        Builds a URI for an protobuf generated code Service Descriptor.
+        @param resourceId The resource id.
```
Comment (matthewd0123) [inline uprotocol/uri/factory/uri_factory.py:35]: match underscore var name here

File: uprotocol/uri/factory/uri_factory.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        @param authorityName The authority name.
```
Comment (matthewd0123) [inline uprotocol/uri/factory/uri_factory.py:36]: match underscore var name here

**Review by matthewd0123 — CHANGES_REQUESTED** [review]
Looks great! Added a few comments, mostly just matching method docs with the arg names.

File: uprotocol/uri/factory/uri_factory.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    """
+    URI Factory that builds URIs from protos
+    """
+
+    @staticmethod
+    def from_proto(
+        service_descriptor: Optional[ServiceDescriptor], resource_id: int, authority_name: Optional[str]
+    ) -> UUri:
+        """
+        Builds a URI for a protobuf generated code Service Descriptor.
```
Comment (princemac) [inline uprotocol/uri/factory/uri_factory.py:34]: Missing parameter service_descriptor in docstring

File: uprotocol/uri/serializer/uriserializer.py [hunk 8]
```diff
@@ -1,52 +1,128 @@
...
     Each transport supports different
     serialization formats.
     """
 
-    @abstractmethod
-    def deserialize(self, uri: T) -> UUri:
+    # The wildcard id for a field.
+    WILDCARD_ID = 0xFFFF
+
+    def serialize(self, uri: Optional[UUri]) -> str:
```
Comment (princemac) [inline uprotocol/uri/serializer/uriserializer.py:32]: It should be static method.

File: uprotocol/uri/serializer/uriserializer.py [hunk 9]
```diff
@@ -1,52 +1,128 @@
...
+        ur_id = None
+        ue_id = int(uri_parts[1], 16)
+        if number_of_parts_in_uri > 2:
+            ue_version = int(uri_parts[2], 16)
+            if number_of_parts_in_uri > 3:
+                ur_id = int(uri_parts[3], 16)
+
+        return ue_id, ue_version, ur_id
+
+    def _build_remote_uri(self, uri_parts, number_of_parts_in_uri):
```
Comment (princemac) [inline uprotocol/uri/serializer/uriserializer.py:68]: It should be static method.

File: uprotocol/uri/serializer/uriserializer.py [hunk 10]
```diff
@@ -1,52 +1,128 @@
...
+
+        sb.append("/")
+        sb.append(hex(uri.ue_id)[2:].upper())
+        sb.append("/")
+        sb.append(hex(uri.ue_version_major)[2:].upper())
+        sb.append("/")
+        sb.append(hex(uri.resource_id)[2:].upper())
+        return re.sub("/+$", "", "".join(sb))
+
+    def _build_local_uri(self, uri_parts, number_of_parts_in_uri):
```
Comment (princemac) [inline uprotocol/uri/serializer/uriserializer.py:57]: It should be static method

File: uprotocol/uri/serializer/uriserializer.py [hunk 11]
```diff
@@ -1,52 +1,128 @@
...
+            if number_of_parts_in_uri > 4:
+                ue_version = int(uri_parts[4], 16)
+                if number_of_parts_in_uri > 5:
+                    ur_id = int(uri_parts[5], 16)
+
+        return auth_name, ue_id, ue_version, ur_id
 
-    @abstractmethod
-    def serialize(self, uri: UUri) -> T:
+    def _build_uri(self, is_local, uri_parts, number_of_parts_in_uri):
```
Comment (princemac) [inline uprotocol/uri/serializer/uriserializer.py:82]: As above method will become static. Then this should also be static.

File: uprotocol/uri/serializer/uriserializer.py [hunk 12]
```diff
@@ -1,52 +1,128 @@
...
+                return UUri()
+            auth_name, ue_id, ue_version, ur_id = self._build_remote_uri(uri_parts, number_of_parts_in_uri)
+        return UUri(
+            authority_name=auth_name,
+            ue_id=ue_id,
+            ue_version_major=ue_version,
+            resource_id=ur_id,
+        )
+
+    def deserialize(self, uri: Optional[str]) -> UUri:
```
Comment (princemac) [inline uprotocol/uri/serializer/uriserializer.py:98]: This can also be converted to static after converting above methods as static.

**Review by princemac — CHANGES_REQUESTED** [review]

**Review by princemac — APPROVED** [review]

File: uprotocol/transport/builder/umessagebuilder.py [hunk 13]
```diff
@@ -0,0 +1,318 @@
...
+        """
+        if request is None:
+            raise ValueError(REQUEST_ERROR)
+        return (
+            UMessageBuilder(
+                request.sink,
+                Factories.UPROTOCOL.create(),
+                UMessageType.UMESSAGE_TYPE_RESPONSE,
+            )
+            .with_priority(UPriority.UPRIORITY_CS4)
```
Comment (stevenhartley) [inline uprotocol/transport/builder/umessagebuilder.py:145]:
```suggestion
            .with_priority(request.priority)
```

**Review by stevenhartley — CHANGES_REQUESTED** [review]
minor code change is all I found

File: uprotocol/transport/builder/umessagebuilder.py (hunk 13 above)
Comment (neelam-kushwah) [reply uprotocol/transport/builder/umessagebuilder.py]: Fixed it and added a test case to verify this scenario.

**Review by stevenhartley — APPROVED** [review]
LGTM, thank you!

**Pull request merged**